### PR TITLE
Convert security-secrets-setup to use common bootstrap package

### PR DESCRIPTION
**Pull request by michaelestrin**
Major refactor.

Fixes: https://github.com/edgexfoundry/edgex-go/issues/2104

### CODE REVIEW

**Comment by michaelestrin** [comment]
This draft PR represents a significant refactoring of the security-secrets-setup service in order to enable it to use the common bootstrap package.  I removed at least two layers of abstraction, separated individual command implementations (based on GoF command pattern), and rewrote the tests.  While I believe I've maintained the functionality and spirit of the prior implementation, I'll let you be the judge.

I'm not able to test this service in situ.  When I built a local image, substituted it into the latest developer-scripts docker-compose file, and brought the system up, it looks to me as if the service crashes.  Please don't blame me.  I have the exact same experience when I use the code from the tip of master.

Tingyu reached out to me earlier today asking about broken tests for this service during `make test` on his local machines (he apparently tried both Windows and Linux).  While I'm not sure what that's all about, I can say it's not my code (since it hasn't been merged).  It also begets the question -- how did changes that cause failing tests get past the build system?

In any case, I'd be grateful if you can expedite a review of these changes.  Happy to answer any questions.  Changes to this service on master to correct the existing issue would likely create much grief for this effort.  Open to someone working interactively with me to help solve the outstanding issues with the master version of this service by making changes to my working branch.

@tsconn23 @bnevis-i @tingyuz @jim-wang-intel

**Comment by tingyuz** [comment]
The failure of test cases I had yesterday was related to configuration of zMQ. Once I configured them correctly all the test cases passed. 

If the build on top of the tip of the master crashes I think it is better to find out the root cause and fix it before pushing more into the branch which will make debugging more difficult.

**Comment by michaelestrin** [comment]
> The failure of test cases I had yesterday was related to configuration of zMQ. Once I configured them correctly all the test cases passed.

I think it's safe to say your test failure experience is unrelated to the problem with the master version of this service crashing.



> If the build on top of the tip of the master crashes I think it is better to find out the root cause and fix it before pushing more into the branch which will make debugging more difficult.

I'm fine with this as long as the changes to actual code in master for this service are very limited.  If they're not, we need to have a conversation on how best to proceed/reconcile.

**Comment by bnevis-i** [comment]
For what is worth, I am observing no crashes of security-secrets-setup in either unit tests nor a full-built container with the nightly docker-compose.   Proceeding to review code.

**Comment by michaelestrin** [comment]
Remove draft status to enable review.  If this is accepted, my intent is to squash everything into a single commit and force-push to keep master commit history clean.

**Comment by jim-wang-intel** [comment]
@michaelestrin I need more time to review this..please hold on the merging.

**Comment by michaelestrin** [comment]
@jim-wang-intel -- I will wait for your review approval.

**Comment by tingyuz** [comment]
Verified and confirmed the master build has no crash and it works well currently so what Michael saw could be the local testing environment issue.

File: internal/security/secrets/command/cache/cache.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+	loggingClient logger.LoggingClient,
+	configuration *config.ConfigurationStruct,
+	generate *generate.Command) (*Command, *flag.FlagSet) {
+
+	return &Command{
+			loggingClient: loggingClient,
+			configuration: configuration,
+			generate:      generate,
+		},
+		flags.flagSet
```
[the remark this answers is not in the document]
Comment (michaelestrin) [reply internal/security/secrets/command/cache/cache.go]: It's implemented this way to provide a way to return the flagset specific to the selected command to be used in a generic way.  Further, the legacy command implementation requires a reference to the flagset to be able to get the configfile option.

I suppose we could set the generic flagset variable in init's Handler() method directly.

**Comment by michaelestrin** [comment]
Update based on requested changes forthcoming.

File: internal/security/secrets/command/cache/cache.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jim-wang-intel) [reply internal/security/secrets/command/cache/cache.go]: I see.  Eventually, I think we are going to remove the `legacy` portion but that is later.  So I am good for now.  Maybe put comments like you said above is a good idea.

[the remark this answers is not in the document]
Comment (michaelestrin) [reply internal/security/secrets/command/cache/cache.go]: Refactored.  Based on updated code, believe pattern of having command factory functions return the flag.FlagSet makes sense.  See updated code for specifics.

**Comment by michaelestrin** [comment]
Rebased.  Addressed comments.  If/when approved, will use GitHub tool to squash and merge (closing #2150)

File: internal/security/secrets/certificates/tls.go [hunk 2]
```diff
@@ -130,19 +132,19 @@ func (gen tlsCertGenerator) Generate() (err error) {
...
 
-	err = gen.writer.Write(gen.seed.TLSKeyFile, pem.EncodeToMemory(&pem.Block{Type: "PRIVATE KEY", Bytes: skPKCS8}), 0600)
+	err = gen.writer.Write(gen.certificateSeed.TLSKeyFile, pem.EncodeToMemory(&pem.Block{Type: "PRIVATE KEY", Bytes: skPKCS8}), 0600)
 	if err != nil {
 		return fmt.Errorf("failed to save TLS server private key: %s", err.Error())
 	}
 
-	gen.logger.Debug(fmt.Sprintf("Saving Root CA certificate to PEM file: %s", gen.seed.CACertFile))
-	err = gen.writer.Write(gen.seed.TLSCertFile, pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: tlsDER}), 0644)
+	gen.logger.Debug(fmt.Sprintf("Saving TLS server certificate to PEM file: %s", gen.certificateSeed.CACertFile))
```
Comment (jim-wang-intel) [inline internal/security/secrets/certificates/tls.go]: the string to log should be `gen.certificatedSeed.TLSCertFile`, not `gen.certificateSeed.CACertFile`

Comment (michaelestrin) [reply internal/security/secrets/certificates/tls.go]: Updated.

**Pull request merged**